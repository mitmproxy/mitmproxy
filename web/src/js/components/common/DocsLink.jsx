import React from "react"
import PropTypes from "prop-types"

DocsLink.propTypes = {
    resource: PropTypes.string.isRequired,
}

export default function DocsLink({ children, resource }) {
    let url = `https://docs.mitmproxy.org/stable/${resource}`
    return (
        <a target="_blank" href={url}>
            {children || <i className="fa fa-question-circle"></i>}
        </a>
    )
}
