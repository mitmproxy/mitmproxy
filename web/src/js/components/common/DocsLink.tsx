import * as React from "react";
type DocLinkProps = {
    children?: React.ReactNode;
    resource: string;
};

export default function DocsLink({ children, resource }: DocLinkProps) {
    let url = `https://docs.mitmproxy.org/stable/${resource}`;
    return (
        <a target="_blank" href={url}>
            {children || <i className="fa fa-question-circle"></i>}
        </a>
    );
}
