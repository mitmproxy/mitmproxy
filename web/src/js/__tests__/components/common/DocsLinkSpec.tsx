import * as React from "react";
import renderer from "react-test-renderer";
import DocsLink from "../../../components/common/DocsLink";

describe("DocsLink Component", () => {
    it("should be able to be rendered with children nodes", () => {
        const docsLink = renderer.create(
            <DocsLink resource="bar">foo</DocsLink>,
        );
        const tree = docsLink.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should be able to be rendered without children nodes", () => {
        const docsLink = renderer.create(<DocsLink resource="bar"></DocsLink>);
        const tree = docsLink.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
