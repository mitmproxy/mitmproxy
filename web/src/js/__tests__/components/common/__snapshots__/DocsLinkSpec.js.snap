// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocsLink Component should be able to be rendered with children nodes 1`] = `
<a
  href="https://docs.mitmproxy.org/stable/bar"
  target="_blank"
>
  foo
</a>
`;

exports[`DocsLink Component should be able to be rendered without children nodes 1`] = `
<a
  href="https://docs.mitmproxy.org/stable/bar"
  target="_blank"
>
  <i
    className="fa fa-question-circle"
  />
</a>
`;
